client/protocols/hinting.py

```diff
@@ -122,7 +119,7 @@ def run(self, messages):
                 log.info('Prompting for hint on %s', question)
                 try:
                     response = self.query_server(messages, question)
-                except (urllib.error.URLError, urllib.error.HTTPError, socket.timeout):
+                except (requests.exceptions.RequestException, requests.exceptions.BaseHTTPError):
```

Comment:
For the future: socket.timeout is not necessary and requests doesn't catch it - it's most likely [a requests bug](https://github.com/kennethreitz/requests/issues/1236)).